src/lib/network_pool/transacation_pool_reorg_tests.ml

```diff
@@ -0,0 +1,433 @@
+            ~removed_commands:(List.map ~f:mk_with_status (prefix @ minor))
+            ~best_tip_ledger:(Option.value_exn test.txn_pool.best_tip_ledger) test.txn_pool ;
+
+          Async.printf "Pool state: \n";
+          let pool_state = Test.Resource_pool.get_all test.txn_pool |> 
+           List.map ~f:(fun tx ->
+            let data = Transaction_hash.User_command_with_valid_signature.data tx in
+            let nonce = data |> User_command.forget_check
+             |> User_command.applicable_at_nonce |> Unsigned.UInt32.to_int in 
+             let fee_payer_pk = data |> User_command.forget_check
+             |> User_command.fee_payer |> Account_id.public_key
+             in
+             (fee_payer_pk,nonce)
+            ) in 
+
+          let log_pool_content = List.map pool_state ~f:(fun (fee_payer_pk,nonce) -> 
+            `String ( Printf.sprintf !"%{sexp: Public_key.Compressed.t} : %d" fee_payer_pk nonce )
+          ) in
+          
+          [%log info] "Pool state" ~metadata:[
+            ("pool state", `List log_pool_content )
+          ];
+
+          let assert_pool_contains pool_state (pk,nonce)= 
+            let actual_opt = List.find pool_state ~f:(fun (fee_payer_pk, actual_nonce ) -> 
+              Public_key.Compressed.equal pk fee_payer_pk &&
+              Int.equal actual_nonce nonce
+            ) in
+          match actual_opt with 
+            | Some actual -> [%test_eq: (Public_key.Compressed.t * int)] (pk,nonce) actual;
+            | None -> failwithf !"Expected transaction from %{sexp: Public_key.Compressed.t} with nonce %d not found \n" pk nonce ();
+          
+          in
+
+          let assert_pool_doesn't_contain pool_state (pk,nonce)= 
+            let actual_opt = List.find pool_state ~f:(fun (fee_payer_pk, actual_nonce ) -> 
+                Public_key.Compressed.equal pk fee_payer_pk &&
+                Int.equal actual_nonce nonce
+              ) in
+              match actual_opt with 
+              | Some _ -> failwithf !"Unexpected transaction from %{sexp: Public_key.Compressed.t} with nonce %d found \n" pk nonce ();
+              | None -> () 
+            
+        in
+        
+        
+        Array.iter minor_specs ~f:(fun (spec:Command_spec.t) ->
+            let sender = Command_spec.sender spec in
+            let (pk, nonce) = Account_spec.to_key_and_nonce sender in
+           
+            let find_owned (acc:Account_spec.t) (txs:Command_spec.t array) = 
+              Array.filter txs ~f:(fun x -> 
+                let sender = Command_spec.sender x in
+                Int.equal acc.key_idx sender.key_idx && 
+                Int.(>) acc.nonce sender.nonce)
+            
+            in  
+            let account_spec = Array.find major_account_spec ~f:(fun spec -> 
+                Int.equal sender.key_idx spec.key_idx && spec.nonce > 0
+              )
+            in
+            match account_spec with
+              | Some account_spec -> 
```

Comment:
Extract `sent_blocking_zkapp`, `total_cost` out of match construction (put it before).

Rewrite `match` without ifs:

```
| Some account_spec when sender.nonce < account_spec.nonce -> (..)
| Some account_spec when sent_blocking_zkapp major_specs pk -> (..)
| Some account_spec when account_spec.balance > total_cost  -> (..)
| Some account_spec -> (..)
| None -> (..)
```